### PR TITLE
Feature/add option to include unpublished objects

### PR DESCRIPTION
**Pull request by clemenskapelle**
Will add a new parameter for the config to add unpublished objects and a new meta information "is-published".

We will need a new version so we won't kill older projects which use this.

### CODE REVIEW

File: Service/Xml.php [hunk 1]
```diff
@@ -48,6 +49,11 @@ public function setIncludeVariants(bool $includeVariants): void
         $this->includeVariants = $includeVariants;
     }
 
+    public function setIncludeUnpublishedProducts(bool $includeUnpublishedProducts): void
```
Comment (mixahead) [inline Service/Xml.php]: 1. phpDocs missing
2. its not only for "Products". The name of function and vars should only be **$includeUnpublished**
3. the function **isIncludeUnpublished** is missing

File: Service/Xml.php [hunk 2]
```diff
@@ -130,7 +136,7 @@ private function exportObject(DataObject $object, $useRecursion=true, $addFields
         $childDataList = [];
 
         if ($useRecursion) {
-            $children = $object->getChildren();
+            $children = $object->getChildren(['object', 'folder'], $this->includeUnpublishedProducts);
```
Comment (mixahead) [inline Service/Xml.php]: Please use Pimcore CONST (e.g. line 152)

File: Service/Xml.php (hunk 1 above)
Comment (clemenskapelle) [reply Service/Xml.php]: done!

File: Service/Xml.php (hunk 2 above)
Comment (clemenskapelle) [reply Service/Xml.php]: done!

**Review by mixahead — APPROVED** [review]

**Pull request merged**